core/src/main/scala/org/typelevel/sbt/TypelevelPlugin.scala

```diff
@@ -64,7 +67,14 @@ object TypelevelPlugin extends AutoPlugin {
     licenses += License.Apache2,
     tlCiHeaderCheck := true,
     tlCiScalafmtCheck := true,
-    tlCiReleaseBranches := Seq("main"),
+    tlCiReleaseBranches := {
+      val central =
+        sonatypeCredentialHost.value.equalsIgnoreCase(Sonatype.sonatypeCentralHost)
+      if (central && tlUntaggedAreSnapshots.value)
+        Seq() // Sonatype Central does not support snapshots
```

Comment:
Do you mind adding a warning log that user should change their sbt settings since this combination is not supported? Perhaps linking to sonatype central docs for reference?

Reply:
@majk-p the tricky thing is that this combination _is_ supported and a user doesn't have to change their sbt settings. It's totally valid for the user to want `-SNAPSHOT` versions and not publish them in CI.

The only combination that's unsupported is attempting to publish `-SNAPSHOT`s to the new central post.

Reply:
> The only combination that's unsupported is attempting to publish `-SNAPSHOT`s to the new central post.

Let's add a warning lot for that, is that what you meant?

Reply:
Yes, that's what I meant, sorry. Perhaps if saying it's unsupported, the log could sat the behavior is likely unexpected and no snapshot will be published to central

Reply:
> Do you mind adding a warning log that user should change their sbt settings since this combination is not supported?

Actually, it's not clear to me where to put the warning 🤔 

We don't want to put the warning in `tlCiReleaseBranches` because there's nothing invalid happening there. It's only if the user overrides its value that we may need to issue a warning, but I don't know how to hook into that esp. when loading the build. I think I might just make this a follow-up issue.